web3_proxy/src/rpcs/request.rs

```diff
@@ -27,7 +27,7 @@ pub enum OpenRequestResult {
 
 /// Make RPC requests through this handle and drop it when you are done.
 /// Opening this handle checks rate limits. Developers, try to keep opening a handle and using it as close together as possible
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
TODO, not sure if this is a good idea, this was needed for the handle, could be a good idea to refactor as well

Reply:
I wouldn't be too afraid of clones. Once it works we can do some profiling and make some flamegraphs and see if it's slow. A lot of times the compiler is smart about optimizing clones away